File: d1.jl
pwd()
println("hello")
println("hello")

function r(filename)
    numbers = Int64[]

    open("d1input.txt") do file
        for line in eachline(file)
            num = parse(Int, line)
            push!(numbers, num)
        end
    end 

    numbers
end

function r2()
    numbers = r("d1input")
    occured = Set{Int64}()
    current = 0

    while true
        for item in numbers
            new_value = current + item
            
            if new_value in occured 
                return new_value
            end
            current = new_value
            push!(occured, new_value)
        end
        println("loop")
    end
end

first_value = r2()
println("first value: $(first_value)")

File: d1.py
def test():
    with open('d1input.txt') as f:
        data = f.readlines()
        data = map(int, data)

    current = 0
    occured = set()

    data = list(data)

    while True:
        for line in data:
            new = current + line

            if new in occured:
                print('occured before!', new)
                return

            current = new
            occured.add(new)


test()

